src/components/ChatEntry.js

```diff
@@ -1,22 +1,35 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp.js';
+
 
 const ChatEntry = (props) => {
+  const localSender = props.sender === 'Vladimir';
+
+  const handleClick = () => {
+    props.onLike();
+  }
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={`chat-entry ${localSender ? 'local' : 'remote'}`}>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"><TimeStamp time = {props.timeStamp}/></p>
+        <button className="like" onClick={handleClick}>
+          {props.liked ? '❤️' : '🤍'}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
 };
```

Comment:
Nice job with prop types here! I'd recommend including the `onLike` function in your prop type validations since it is essential for the like functionality to work.